### PR TITLE
Add definition of Undetermined to SSC overview

### PR DESCRIPTION
**Pull request by armchairlinguist**
We get a fair number of questions about this term, and previously it was only defined in the filter descriptions. This PR adds it to the overview page along with the other possibilities.

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team

### CODE REVIEW

**Review by s-santillan — APPROVED** [review]
Thank you!

File: docs/semgrep-supply-chain/overview.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ Semgrep Supply Chain generates a **finding** when it detects a match. If the dep
 
 A finding is **unreachable** if the dependency contains a known vulnerability, but the vulnerable matching code is not used in your codebase.
 
+A finding is **undetermined** if Semgrep detects a match for the dependency in the vulnerable version range, but does not scan for the reachability of this vulnerability. This is most common with older or lower severity vulnerabilities.
```
Comment (andy-r2c) [inline docs/semgrep-supply-chain/overview.md:43]: I think we can also just say older and lower severity vulnerabilities don't have reachability rules? I feel like some users might interpret "does not scan for the reachability of this vulnerability" as a parsing/code issue

Comment (armchairlinguist) [reply docs/semgrep-supply-chain/overview.md:43]: Would you prefer:

A finding is **undetermined** if Semgrep detects a match for the dependency in the vulnerable version range, but does not have a reachability rule for this vulnerability.

or something else? Open to suggestions!

Comment (armchairlinguist) [reply docs/semgrep-supply-chain/overview.md:43]: @andy-r2c got a sec to take a look at this alternative?

Comment (andy-r2c) [reply docs/semgrep-supply-chain/overview.md:43]: "A finding is undetermined if Semgrep detects a match for the dependency in the vulnerable version range, but does not have a reachability rule for this vulnerability."

works perfectly!

**Review by andy-r2c — APPROVED** [review]

**Pull request merged**